core/server/api_container/server/service_network/service_identifiers/service_identifier.go

```diff
@@ -0,0 +1,83 @@
+package service_identifiers
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/uuid_generator"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+const (
+	uuidKey             = "uuid"
+	shortenedUuidStrKey = "shortened_uuid_str"
+	nameKey             = "name"
+)
+
+type serviceIdentifier struct {
+	uuid             service.ServiceUUID
+	shortenedUuidStr string
+	name             service.ServiceName
+}
+
+func NewServiceIdentifier(uuid service.ServiceUUID, name service.ServiceName) *serviceIdentifier {
+	uuidStr := string(uuid)
+	shortenedUuidStr := uuid_generator.ShortenedUUIDString(uuidStr)
+
+	return &serviceIdentifier{uuid: uuid, shortenedUuidStr: shortenedUuidStr, name: name}
+}
+
+func (serviceIdentifier *serviceIdentifier) GetUuid() service.ServiceUUID {
+	return serviceIdentifier.uuid
+}
+
+func (serviceIdentifier *serviceIdentifier) GetShortenedUUIDStr() string {
+	return serviceIdentifier.shortenedUuidStr
+}
+
+func (serviceIdentifier *serviceIdentifier) GetName() service.ServiceName {
+	return serviceIdentifier.name
+}
+
+func (serviceIdentifier *serviceIdentifier) MarshalJSON() ([]byte, error) {
```

Comment:
Given that `serviceIdentifier` is a private struct, can we drop this in favor of the default implementation of `json.Marshal`?

Reply:
if we do that we are going to break the encapsulation.
<img width="652" alt="serviceNane" src="https://github.com/kurtosis-tech/kurtosis/assets/29951623/7c93314a-eba9-4590-94c9-2d802c0beb2b">

I could implement the `privateInternalStruct` pattern that you show me today instead, but in this case, we will need a custom Marshal and Unmarshal method also. WDYT?

Reply:
Fine to leave as is then!